### PR TITLE
Feat easing type

### PR DESCRIPTION
**Pull request by DirectuX**
Hi Petr, 
can you try this version ? 

try holding 'UP' more or less long time
try to reach bubbles at different speed (burst/no burst)

### CODE REVIEW

File: src/game_objects/player.tbasicu [hunk 1]
```diff
@@ -108,25 +114,76 @@ type goPlayer
...
+      oldGameTime = 0
+      me.thrustImpulse = %FALSE
+    else
+      if me.thrust < 6 Then ' speed is low enough to not burst bubbles
+        function = %TRUE
+      endif
+      oldGameTime = context.gametime ' store current time for diff calculation
+      easeCache = (me.easeSimple(temps, currentCharge , currentThrustDuration) + crumb)
+      crumb = frac(easeCache) ' maybe not useful in the end ?
+      me.thrust = fix(easeCache)
```
Comment (DirectuX) [inline src/game_objects/player.tbasicu]: I wanted to displace the division by frameRate from game.tbasicu ( line 194 in the newest file ) to here. But it didn't work. I don't know why.

File: src/BetweenTheBubbles.tbasic [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿' [+] New inifinte bubbleverse!
+' [+] New inifinte bubbleverse!
```
Comment (petrSchreiber) [inline src/BetweenTheBubbles.tbasic:1]: I am not sure I follow changes in this file - are they intentional, please?

File: src/game_objects/player.tbasicu (hunk 1 above)
Comment (petrSchreiber) [reply src/game_objects/player.tbasicu]: Strange - could you please post a version which did not work in a reply to this?

File: src/BetweenTheBubbles.tbasic (hunk 2 above)
Comment (DirectuX) [reply src/BetweenTheBubbles.tbasic:1]: Just re-save without bom nor the thinAir unwanted chars at the end of script. That's the unwanted result. I can't do nothing about it.

Comment (petrSchreiber) [reply src/BetweenTheBubbles.tbasic:1]: I see, thank you for the explanation 👍

File: src/game_objects/player.tbasicu (hunk 1 above)
Comment (DirectuX) [reply src/game_objects/player.tbasicu]: see [this draft](https://github.com/petrSchreiber/BetweenTheBubbles-tb/pull/8)

**Comment by petrSchreiber** [comment]
Hi Sebastian,

I tried code from this branch, and while the bug does move legs indeed, even manages to burst the bubble somehow, it basically does not move at all.

**Comment by DirectuX** [comment]
[This ](https://github.com/petrSchreiber/BetweenTheBubbles-tb/blob/feat_easingType/Projet%20sans%20nom.ogg)is how it plays on my computer. Still a draft.
Todo: 
- reduce reload time by making charging rate not linear, perhaps sqr ?
- create a visual feedback of charging

Are you sure  you did _Clone or Download_ [this ](https://github.com/petrSchreiber/BetweenTheBubbles-tb/tree/feat_easingType)version ?

File: src/game_objects/player.tbasicu [hunk 3]
```diff
@@ -107,26 +130,99 @@ type goPlayer
...
+      if me.thrust < %MIN_BURST_SPEED Then function = %TRUE ' speed is low enough to not burst bubbles
+      easeCache = (me.easeSimple(me.temps, me.currentCharge , me.currentThrustDuration) + me.crumb)
+      me.crumb = frac(easeCache) ' maybe not useful in the end ?
+      me.thrust = fix(easeCache)
+      me.thrustImpulse = %TRUE
+    endif
+
+  end function
+'[todo] First lvl : The Pilgrim brake to a stop between each jump. (charge - jump - charge - jump ...)
+'       Feat unlock : the Pilgrim can tie each jump to the previous (like swimming : charge - jump and charge while gliding along - jump and charge while gliding along - jump and charge while gliding along)
```
Comment (DirectuX) [inline src/game_objects/player.tbasicu:181]: if you're ok with that, I would prefer work on this in another feat_branch while merging this one.

**Comment by petrSchreiber** [comment]
My bad! I tried other branch with the same name. I do like this one :)

Maybe some indicator of "charge" would be nice, but that could be another PR :)

File: src/game_objects/player.tbasicu (hunk 3 above)
Comment (petrSchreiber) [reply src/game_objects/player.tbasicu:181]: Sure!

**Review by petrSchreiber — APPROVED** [review]
Hi Sebastian,

I approve the changes, feel free to "Squash and merge" :)

**Pull request merged**